### PR TITLE
:bug: When infrastructureRef is nil, set InfrastructureReadyCondition to true

### PR DESCRIPTION
**Pull request by vincepri**
… to true

Some clients, like MachineHealthCheck are checking this condition to continue operations. Set it for backward compatibility

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/area api

### CODE REVIEW

**Comment by sbueringer** [comment]
Chnage makes sense to me

**Comment by fabriziopandini** [comment]
lgtm pending fixing the linter error

File: internal/controllers/cluster/cluster_controller_test.go [hunk 1]
```diff
@@ -172,6 +172,50 @@ func TestClusterReconciler(t *testing.T) {
 		}, timeout).Should(BeTrue())
 	})
 
+	t.Run("Should successfully patch a cluster object if the spec diff is empty but the status conditions diff is not", func(t *testing.T) {
```
Comment (sbueringer) [inline internal/controllers/cluster/cluster_controller_test.go:175]: Can you provide some context what use case this test is testing?

Comment (sbueringer) [reply internal/controllers/cluster/cluster_controller_test.go:175]: This seems to be some other controller also patching InfrastructureReady to true in parallel to the cluster controller?

File: internal/controllers/cluster/cluster_controller_test.go [hunk 2]
```diff
@@ -172,6 +172,50 @@ func TestClusterReconciler(t *testing.T) {
...
+				return false
+			}
+			return len(cluster.Finalizers) > 0
+		}, timeout).Should(BeTrue())
+
+		// Patch
+		g.Eventually(func() bool {
+			ph, err := patch.NewHelper(cluster, env)
+			g.Expect(err).ToNot(HaveOccurred())
+			conditions.MarkTrue(cluster, clusterv1.InfrastructureReadyCondition)
```
Comment (sbueringer) [inline internal/controllers/cluster/cluster_controller_test.go:204]: Is this a diff? Isn't the cluster controller also setting this condition to true in reconcile?

File: internal/controllers/cluster/cluster_controller_test.go (hunk 1 above)
Comment (vincepri) [reply internal/controllers/cluster/cluster_controller_test.go:175]: I only added it for completeness, seems this really tests the patch helper?

Comment (vincepri) [reply internal/controllers/cluster/cluster_controller_test.go:175]: The same test, but for `cluster.Status.InfrastructureReady = true` is above this one

File: internal/controllers/cluster/cluster_controller_test.go (hunk 2 above)
Comment (vincepri) [reply internal/controllers/cluster/cluster_controller_test.go:204]: Is the controller running? Wouldn't the above patch also be the same?

Comment (sbueringer) [reply internal/controllers/cluster/cluster_controller_test.go:204]: l.192 seems to suggest that the controller is running :)

File: internal/controllers/cluster/cluster_controller_test.go (hunk 1 above)
Comment (sbueringer) [reply internal/controllers/cluster/cluster_controller_test.go:175]: Ah, I didn't realize this is ~ a copy of an existing test :D

**Comment by sbueringer** [comment]
Let's merge this PR to unblock the definitely good & useful change. Let's continue to discuss the unit test and then follow-up if necessart

/lgtm
/approve

File: internal/controllers/cluster/cluster_controller_test.go (hunk 2 above)
Comment (sbueringer) [reply internal/controllers/cluster/cluster_controller_test.go:204]: Ran both tests through the debugger. The patch here is literally a no-op (same in the pre-existing test)

**Pull request merged**